### PR TITLE
Audit access tokens + Requests through access tokens contain the user_id in their audits

### PR DESCRIPTION
**Pull request by Martouta**
Closes [THREESCALE-421](https://issues.redhat.com/browse/THREESCALE-421)

- [X] Add timestamps to access tokens to make it easier for support to debug.
- [X] Add audits for creating/updating/destroying access tokens.
- [X] When a request that contains audits is done by the API through an access token. That audit also contains the user_id as we do for the requests by the UI.

### CODE REVIEW

File: app/models/access_token.rb [hunk 1]
```diff
@@ -5,6 +5,10 @@ class AccessToken < ApplicationRecord
 
   serialize :scopes, Array
 
+  audited only: %i[owner_id scopes name permission created_at updated_at]
+
+  delegate :provider_id_for_audits, to: :owner
```
Comment (guicassolato) [inline app/models/access_token.rb]: `owner` is a polymorphic association of `AccessToken`, which now seems to only have `User` as type in the db of Saas. Just in case, unless or until we are 100% positive that no other type of entity can own an access token, maybe better start with:

```suggestion
  def provider_id_for_audits
    owner.try(:provider_id_for_audits)
  end
```

File: app/lib/api_authentication/by_access_token.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module ByAccessToken
     extend ActiveSupport::Concern
 
     def current_user
-      @current_user ||= authenticated_token&.owner or defined?(super) && super
+      @current_user ||= User.current = authenticated_token&.owner or defined?(super) && super
```
Comment (guicassolato) [inline app/lib/api_authentication/by_access_token.rb]: This seems to belong to a different PR.

Comment (Martouta) [reply app/lib/api_authentication/by_access_token.rb]: What do you mean? that is one of the purposes of the PR. Because the user of the audit is taken from User.current, which used to be `nil` only when the user was identified by access token, for this.

https://github.com/3scale/porta/blob/e6be7e4c686590fc7f537f723ad6567f8b709298/config/initializers/audited_hacks.rb#L180-L185

File: app/models/access_token.rb (hunk 1 above)
Comment (Martouta) [reply app/models/access_token.rb]: Look at this (the `class_name`):
https://github.com/3scale/porta/blob/e6be7e4c686590fc7f537f723ad6567f8b709298/app/models/access_token.rb#L2

And this:
image

File: app/lib/api_authentication/by_access_token.rb (hunk 2 above)
Comment (guicassolato) [reply app/lib/api_authentication/by_access_token.rb]: I mean auditing the access token changes AND adding the user to the audits in general are 2 different unrelated changes IMO, even though the Jira asks both in one same ticket.

File: app/models/access_token.rb (hunk 1 above)
Comment (guicassolato) [reply app/models/access_token.rb]: I don't know why it's polymorphic then 🤷‍♂️

File: app/lib/api_authentication/by_access_token.rb (hunk 2 above)
Comment (Martouta) [reply app/lib/api_authentication/by_access_token.rb]: Aha, I see :+1:

File: app/models/access_token.rb (hunk 1 above)
Comment (Martouta) [reply app/models/access_token.rb]: Me neither :joy: It is just overcomplicating :thinking: Maybe in the past, it needed to be polymorphic for real but not anymore :woman_shrugging:
